### PR TITLE
Allow inclusion of unloaded segments in stats

### PR DESCRIPTION
**Pull request by s1monw**
Today we have no chance to fetch actual segment stats for segments that
are currently unloaded. This is relevant in the case of frozen indices.
This allows to monitor how much memory a frozen index would use if it was
unfrozen.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/index/engine/FrozenEngine.java [hunk 1]
```diff
@@ -79,6 +81,13 @@ public FrozenEngine(EngineConfig config) {
         try (DirectoryReader reader = DirectoryReader.open(directory)) {
             canMatchReader = ElasticsearchDirectoryReader.wrap(new RewriteCachingDirectoryReader(directory, reader.leaves()),
                 config.getShardId());
+            // we record the segment stats here - that's what the reader needs when it's open and it give the user
+            // an idea of what it can save when it's closed
+            this.stats = new SegmentsStats();
+            for (LeafReaderContext ctx : reader.getContext().leaves()) {
+                SegmentReader segmentReader = Lucene.segmentReader(ctx.reader());
+                    fillSegmentStats(segmentReader, true, stats);
```
Comment (dnhatn) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/index/engine/FrozenEngine.java]: nit: indentation

File: server/src/main/java/org/elasticsearch/action/admin/indices/stats/CommonStats.java [hunk 2]
```diff
@@ -207,7 +207,7 @@ public CommonStats(IndicesQueryCache indicesQueryCache, IndexShard indexShard, C
                         completion = indexShard.completionStats(flags.completionDataFields());
                         break;
                     case Segments:
-                        segments = indexShard.segmentStats(flags.includeSegmentFileSizes());
+                        segments = indexShard.segmentStats(flags.includeSegmentFileSizes(),flags.includeUnloadedSegments());
```
Comment (dnhatn) [inline server/src/main/java/org/elasticsearch/action/admin/indices/stats/CommonStats.java]: nit: space between parameters.

**Review by dnhatn — APPROVED** [review]
LGTM. I left two super minor nits.

**Review by ywelsch — APPROVED** [review]
LGTM. This is such useful information that I'm wondering whether we should have it for closed replicated indices as well (by opening the index very shortly to compute the stats when creating the NoOpEngine)

**Comment by s1monw** [comment]
> LGTM. This is such useful information that I'm wondering whether we should have it for closed replicated indices as well (by opening the index very shortly to compute the stats when creating the NoOpEngine)

sounds good to me. I will open a second PR for this.

**Pull request merged**